manager_janitor.go

```diff
@@ -335,6 +335,105 @@ func (mgr *Manager) pindexesRestart(
 	return errs
 }
 
+func (mgr *Manager) activatePIndex(pi *pindexRestartReq) (*PIndex, error) {
+	err := mgr.stopPIndex(pi.pindex, false)
+	if err != nil {
+		log.Printf("janitor: error closing pindex %s", pi.pindex.Name)
+		return nil, err
+	}
+	// rename the pindex folder and name as per the new plan
+	newPath := mgr.PIndexPath(pi.planPIndexName)
+	if newPath != pi.pindex.Path {
+		log.Printf("janitor: proceeding to rename %s to %s", pi.pindex.Path, newPath)
+		err := os.Rename(pi.pindex.Path, newPath)
+		if err != nil {
+			cleanDir(pi.pindex.Path)
```

Comment:
What is the reasoning behind deleting all the older pindex dir paths? 
Where does that leave the system?
Check this wherever you are following the same pattern?